packages/fxa-settings/src/pages/ResetPassword/ResetPasswordConfirmed/container.tsx

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useLocation } from '@reach/router';
+
+import { useNavigateWithQuery } from '../../../lib/hooks/useNavigateWithQuery';
+import ResetPasswordConfirmed from '.';
+import { MozServices } from '../../../lib/types';
+import {
+  Integration,
+  isOAuthIntegration,
+  useAuthClient,
+  useSensitiveDataClient,
+} from '../../../models';
+import { useFinishOAuthFlowHandler } from '../../../lib/oauth/hooks';
+import OAuthDataError from '../../../components/OAuthDataError';
+import { hardNavigate } from 'fxa-react/lib/utils';
+import { AuthUiErrors } from '../../../lib/auth-errors/auth-errors';
+import { AuthError } from '../../../lib/oauth';
+import { useState } from 'react';
+import GleanMetrics from '../../../lib/glean';
+
+const ResetPasswordConfirmedContainer = ({
+  integration,
+  serviceName,
+}: {
+  integration: Integration;
+  serviceName: MozServices;
+} & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+  const location = useLocation();
+  const navigateWithQuery = useNavigateWithQuery();
+  const queryParams = location.search;
+  const sensitiveDataClient = useSensitiveDataClient();
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const { uid, sessionToken, keyFetchToken, unwrapBKey, verified } =
+    sensitiveDataClient.getData('accountResetData');
+
+  const email = sensitiveDataClient.getData('email') || '';
+
+  const handleOAuthRedirectError = (error: AuthError) => {
+    if (
+      error.errno === AuthUiErrors.TOTP_REQUIRED.errno ||
+      error.errno === AuthUiErrors.INSUFFICIENT_ACR_VALUES.errno
+    ) {
+      navigateWithQuery(`/inline_totp_setup${queryParams}`, {
+        state: {
+          email,
+          uid,
+          sessionToken,
+          verified,
+          keyFetchToken,
+          unwrapBKey,
```

Comment:
iirc the reason we created the sensitive data client just so we don't pass certain value this way.

Reply:
I'll file a follow up to update how the InlineTotpSetup component receives this data.